stories/packages/BooleanQueryTokenizer.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { storiesOf } from '@storybook/react';
+import { BooleanQueryTokenizer, TextArea } from '@textkernel/oneui';
+
+const tokenizer = new BooleanQueryTokenizer();
+const initialValue =
+    '-php AND +c++ +--+ -\n' +
+    'java* NOT *script\n' +
+    'NEAR (rust OR tex*nel) OR\n' +
+    '-.NET "TO BE OR NOT TO BE"';
+
+const columnStyle = {
+    width: '30%',
+    padding: '1em',
+};
+
+storiesOf('packages|BooleanQueryTokenizer', module).add('BooleanQueryTokenizer', () => {
+    const [tokens, setTokens] = React.useState<unknown[]>(tokenizer.tokenize(initialValue));
+    const [wordsAndPhrases, setWordsAndPhrases] = React.useState(
+        tokenizer.extractWordsAndPhrases(initialValue)
+    );
+    const tokenizeBooleanQuery = (event) => {
+        const { value } = event.target;
+        setTokens(tokenizer.tokenize(value));
+        setWordsAndPhrases(tokenizer.extractWordsAndPhrases(value));
+    };
+    return (
+        <div style={{ display: 'flex' }}>
+            <div style={columnStyle}>
+                <h3>Boolean query</h3>
+                <TextArea
+                    context="brand"
+                    defaultValue={initialValue}
+                    disabled={false}
+                    isBlock
+                    onChange={tokenizeBooleanQuery}
+                    placeholder="Enter a boolean query..."
+                    size="normal"
+                    style={{
+                        resize: 'vertical',
+                        height: '100%',
+                    }}
+                />
+            </div>
+            <div style={columnStyle}>
+                <h3>Words and phrases</h3>
+                <pre>{JSON.stringify(wordsAndPhrases, null, 4)}</pre>
+            </div>
+            <div style={columnStyle}>
+                <h3>Tokens</h3>
+                <pre>{JSON.stringify(tokens, null, 4)}</pre>
+            </div>
+        </div>
```

Comment:
Maybe out of scope here, but would be interesting to see how to use this together with the highlighter, especially for none trivial cases. E.g. excluding `NOT xxx` or when using `NEAR` or `java * script` type of things.

Reply:
I think is is out of scope indeed. `BooleanQueryTokenizer` doesn't depend from Higlighter (the other way around maybe, but still there is no hard dependency).